lib/socket.js

```diff
@@ -84,7 +85,7 @@ class Transport extends EventEmitter {
 
 const socketFactory = (connect) => {
   const resultConnectFn = (...options) => {
-    const callback = common.extractCallback(options);
+    const callback = common.cbExtract(options);
```

Comment:
We need to avoid throwing, it terminates process, it's better to bubble an error to upper level without process termination.